File: other/chatGPT/ReverseArray.java

public class ReverseArray {
    
    private static int[] reverseArr(int[] nums){

        if (nums.length < 0 || nums.length == 0) return new int[] {};

        int left = 0;
        int right = nums.length - 1;

        while (right > left){

            int temp = nums[left];
            nums[left] = nums[right];
            nums[right] = temp;
            right --;
            left ++;
        }

        return nums;
    }

    public static void main(String[] args){

        int[] num = new int[] {4,4,5,6,78,964,8,2};

        for (int i : reverseArr(num)){
            System.out.print(i + " ");
        }
    }
}

File: other/chatGPT/ReverseArray.js

const reverseArray = (nums) => {

    let left = 0;
    let right = nums.length - 1;

    while (right > left){

        let temp = nums[left];
        nums[left] = nums[right];
        nums[right] = temp;
        left ++;
        right --;
    }

    return nums;
}

const nums = [153,1,63,1];
for (let i of reverseArray(nums)){
    console.log(i);
}